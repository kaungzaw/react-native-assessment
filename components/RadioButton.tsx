import { Pressable, StyleSheet } from "react-native";
import { ThemedView } from "./ThemedView";
import { ThemedText } from "./ThemedText";

type RadioButtonProps = {
  label: string;
  selected?: boolean;
  onPress: () => void;
};

export default function RadioButton({
  label,
  selected = false,
  onPress,
}: RadioButtonProps) {
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        pressed ? { opacity: 0.5 } : {},
      ]}
      onPress={onPress}
    >
      <ThemedText>{label}</ThemedText>
      <ThemedView style={styles.radio}>
        {selected ? <ThemedView style={styles.radioSelected} /> : null}
      </ThemedView>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "transparent",
    marginHorizontal: 30,
    marginVertical: 14,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  radio: {
    height: 24,
    width: 24,
    borderRadius: 12,
    borderWidth: 3,
    borderColor: "grey",
    alignItems: "center",
    justifyContent: "center",
  },
  radioSelected: {
    height: 12,
    width: 12,
    borderRadius: 6,
    backgroundColor: "grey",
  },
});
