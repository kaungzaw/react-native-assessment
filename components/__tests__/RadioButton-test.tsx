import * as React from "react";
import { render } from "@testing-library/react-native";

import RadioButton from "../RadioButton";

it(`renders correctly`, () => {
  const tree = render(
    <RadioButton label="RadioButton" onPress={() => {}} selected />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

describe("Room", () => {
  test("Room renders correctly", () => {
    const { getByText } = render(
      <RadioButton label="RadioButton" onPress={() => {}} selected />
    );

    getByText("RadioButton");
  });
});
