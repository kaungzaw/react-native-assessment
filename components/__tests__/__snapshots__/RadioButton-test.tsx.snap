// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 30,
        "marginVertical": 14,
      },
      {},
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    RadioButton
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "borderColor": "grey",
          "borderRadius": 12,
          "borderWidth": 3,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "backgroundColor": "grey",
            "borderRadius": 6,
            "height": 12,
            "width": 12,
          },
        ]
      }
    />
  </View>
</View>
`;
